### DISCUSSION TITLE
Switch or Input?

### DISCUSSION THREAD

**Discussion by richardsg307** [arendst/Tasmota#20306] [Q&A] [answered]
I've got a mains sensor that provides an optically isolated signal. Low means the mains is active. It turns on/off over many many seconds so is not a rapid thing.
Should I declare that as a Switch or an Input in the GPIO Configuration?
(I've read the docs and it's not clear to me. Input doesn't even mention Pullups as a option)

**Comment by sfromis** [comment, marked as answer]
Configuring as "Input" does nothing, as this only reserves the pin for usage by Berry, where the Berry code then can choose such details as pullup/down. Switch sounds right for what you describe.
